pkg/storage/watch_cache.go

```diff
@@ -240,8 +245,6 @@ func (w *watchCache) updateCache(resourceVersion uint64, event *watchCacheEvent)
 
 // List returns list of pointers to <storeElement> objects.
 func (w *watchCache) List() []interface{} {
-	w.RLock()
-	defer w.RUnlock()
```

Comment:
It is correct - it calls waitUntilFreshAndBlock and in line 250 you have:
```
// You HAVE TO explicitly call w.RUnlock() after this function.
```

It is to avoid code duplication.